4_visualize/src/plot_results.R

```diff
@@ -0,0 +1,412 @@
+#' @return A munged dataframe of surface, middle, and bottom temperature GLM
+#' predictions for all successfully simulated lakes. The munge table includes 
+#' the columns site_id, gcm, date, month, year DOY, decade, period, gcm, 
+#' ice, hice (ice thickness), evap (evaporation), n_layers (number of simulated 
+#' layers), depth_class (surface, middle, or bottom), depth, and temperature 
+get_surface_middle_bottom_preds <- function(run_groups) {
+  surface_middle_bottom_preds <- run_groups %>%
+    group_by(site_id) %>%
+    group_modify(~ {
+      site_meteo_files <- .x$raw_meteo_fl
+      site_export_files <- .x$export_fl
+      run_gcm_list <- .x$gcm
+      # Pull predictions for each site
+      site_preds <- get_site_preds(site_meteo_files, site_export_files, run_gcm_list)
+      
+      # Munge to long format
+      site_preds_long <- munge_long(site_preds)
+      
+      # Identify surface, middle, and bottom depths for each site and filter
+      # predictions to only those depths
+      plot_depths <- get_plotting_depths(site_preds_long)
+      filtered_preds <- filter(site_preds_long, depth %in% plot_depths) %>%
+        mutate(depth_class=case_when(
+          depth==plot_depths$plot_depth_surface ~ 'surface',
+          depth==plot_depths$plot_depth_middle ~ 'middle',
+          depth==plot_depths$plot_depth_bottom ~ 'bottom'
+        ), .before=depth)
+      return(filtered_preds)
+    })
+  return(surface_middle_bottom_preds)
+}
+
+#' @title Munge GLM predictions from wide to long format
+#' @description pivots wide-format GLM predictions (with columns like
+#' 'temp_0', 'temp_0.5', etc.) to wide format, with a column for 'depth'
+#' and a column for 'temperature'
+#' @param input_wide The wide-formatted input data
+#' @return a munged dataframe of GLM predictions that includes all of 
+#' the columns in the original wide data except the 'temp_X' columns,
+#' which are instead reduced to two columns: 'depth' and 'temperature'
+munge_long <- function(input_wide) {
+  input_long <- input_wide %>%
+    pivot_longer(temp_0:colnames(input_wide)[ncol(input_wide)], names_to="depth", values_to="temperature") %>%
+    mutate(depth = as.numeric(str_remove(depth, 'temp_'))) %>%
+    arrange(date, depth)
+}
+
+#' @title Get the surface, middle, and bottom depths for a given lake, 
+#' for plotting. Modified from Alison's code for picking surface, middle, 
+#' and bottom depths from observational data: 
+#' https://code.usgs.gov/wma/wp/lake-temperature-neural-networks/-/blob/master/3_assess/src/inspect_site.R#L81-93
+#' @description For a given lake, identify which depths represent the
+#' surface, middle, and bottom sections of the lake
+#' @param predictions_long A long-format dataset of GLM predictions at
+#' 0.5-meter intervals
+#' @return A named list of the surface, middle, and bottom depths to be used
+#' to plot data for a given lake.
+get_plotting_depths <- function(predictions_long) {
+  n_lambda <- 1
```

Comment:
Why lambda? Seems like you could get away without this since it isn't parameterizable at all

Reply:
Yup that's an artifact of Alison's original code that I've just been reusing -- will remove